python-lib/plugin_config_loading.py

```diff
@@ -0,0 +1,174 @@
+
+    df = custom_corrections_dataset.get_dataframe(infer_with_pandas=False)
+    df = clean_text_df(df, dropna_columns=[word_column["name"]]).fillna("").astype(str)
+    custom_corrections_dict = {row[0]: row[1] for row in df.itertuples(index=False)}
+    return custom_corrections_dict
+
+
+def load_plugin_config() -> Dict:
+    """Utility function to load, resolve and validate all plugin config into a clean `params` dictionary
+
+    Returns:
+        Dictionary of parameter names (key) and values
+    """
+    params = {}
+    recipe_config = get_recipe_config()
+
+    # input dataset
+    input_dataset_names = get_input_names_for_role("input_dataset")
+    if len(input_dataset_names) == 0:
+        raise PluginParamValidationError("Please specify input dataset")
+    params["input_dataset"] = dataiku.Dataset(input_dataset_names[0])
+
+    # output dataset
+    output_dataset_names = get_output_names_for_role("output_dataset")
+    if len(output_dataset_names) == 0:
+        raise PluginParamValidationError("Please specify output dataset")
+    params["output_dataset"] = dataiku.Dataset(output_dataset_names[0])
+
+    # custom_vocabulary (optional input dataset)
+    params["custom_vocabulary_set"] = set()
+    custom_vocabulary_input = get_input_names_for_role("custom_vocabulary")
+    if len(custom_vocabulary_input) != 0:
+        custom_vocabulary_dataset = dataiku.Dataset(custom_vocabulary_input[0])
+        params["custom_vocabulary_set"] = custom_vocabulary_checker(custom_vocabulary_dataset)
+    logging.info("Custom vocabulary set: {}".format(params["custom_vocabulary_set"]))
+
+    # custom_corrections (optional input dataset)
+    params["custom_corrections"] = {}
+    custom_corrections_input = get_input_names_for_role("custom_corrections")
+    if len(custom_corrections_input) != 0:
+        custom_corrections_dataset = dataiku.Dataset(custom_corrections_input[0])
+        params["custom_corrections"] = custom_corrections_checker(custom_corrections_dataset)
+    logging.info("Custom corrections: {}".format(params["custom_corrections"]))
+
+    # diagnosis dataset (optional output dataset)
+    diagnosis_dataset_names = get_output_names_for_role("diagnosis_dataset")
+    params["diagnosis_dataset"] = None
+    params["compute_diagnosis"] = False
+    if len(diagnosis_dataset_names) != 0:
+        logging.info("Spellchecker diagnosis will be computed")
+        params["compute_diagnosis"] = True
+        params["diagnosis_dataset"] = dataiku.Dataset(diagnosis_dataset_names[0])
+    else:
+        logging.info("Spellchecker diagnosis will not be computed")
+
+    # path to the folder of dictionaries
+    params["dictionary_folder_path"] = os.path.join(get_recipe_resource(), "dictionaries")
+
+    # List of text columns
+    params["text_column"] = recipe_config.get("text_column")
+    logging.info("Text column: {}".format(params["text_column"]))
+    if params["text_column"] == "":
```

Comment:
Need to check that it is a valid column name and also check for both "" and None with "not"
```suggestion
    if not params["text_column"]:
```